dask/dataframe/io/parquet/arrow.py

```diff
@@ -265,149 +91,540 @@ def _index_in_schema(index, schema):
+
+    if has_pandas_metadata:
+        pandas_metadata = json.loads(schema.metadata[b"pandas"].decode("utf8"))
+        (
+            index_names,
+            column_names,
+            storage_name_mapping,
+            column_index_names,
+        ) = _parse_pandas_metadata(pandas_metadata)
+        if categories is None:
+            categories = []
+            for col in pandas_metadata["columns"]:
+                if (col["pandas_type"] == "categorical") and (
+                    col["name"] not in categories
                 ):
-                    dataset.schema = dataset.metadata.schema
-                    dataset.metadata = None
-        else:
-            partitions = []
-
-        # Statistics are currently collected at the row-group level only.
-        # Therefore, we cannot perform filtering with split_row_groups=False.
-        # For "partitioned" datasets, each file (usually) corresponds to a
-        # row-group anyway.
-        # TODO: Map row-group statistics onto file pieces for filtering.
-        #       This shouldn't be difficult if `col_chunk_paths==True`
-        if not split_row_groups and not col_chunk_paths:
-            if gather_statistics is None and not partitions:
-                gather_statistics = False
-                if filters:
-                    raise ValueError(
-                        "Filters not supported with split_row_groups=False "
-                        "(unless proper _metadata is available)."
-                    )
-            if gather_statistics and not partitions:
-                raise ValueError(
-                    "Statistics not supported with split_row_groups=False."
-                    "(unless proper _metadata is available)."
-                )
+                    categories.append(col["name"])
+    else:
+        index_names = []
+        column_names = schema.names
+        storage_name_mapping = {k: k for k in column_names}
+        column_index_names = [None]
+
+    if index is None and index_names:
+        index = index_names
+
+    if set(column_names).intersection(partitions):
+        raise ValueError(
+            "partition(s) should not exist in columns.\n"
+            "categories: {} | partitions: {}".format(column_names, partitions)
+        )
 
-        if dataset.metadata:
-            schema = dataset.metadata.schema.to_arrow_schema()
-        else:
-            schema = dataset.schema.to_arrow_schema()
-        columns = None
+    column_names, index_names = _normalize_index_columns(
+        columns, column_names + partitions, index, index_names
+    )
+
+    all_columns = index_names + column_names
 
-        has_pandas_metadata = (
-            schema.metadata is not None and b"pandas" in schema.metadata
+    # Check that categories are included in columns
+    if categories and not set(categories).intersection(all_columns):
+        raise ValueError(
+            "categories not in available columns.\n"
```

Comment:
(I guess each of these conditions should be tested - I have not yet looked through the tests)

Reply:
We do test for this `ValueError` in `test_categories` - Hopefully that is sufficient

Reply:
ok